appcreds/appcreds.go

```diff
@@ -33,8 +34,32 @@ func New(ctx context.Context, m manipulate.Manipulator, namespace string, name s
 	return NewWithAppCredential(ctx, m, creds)
 }
 
+// NewFromTemplate generates a new CSR for the provided app credential and calls the upstream service using the supplied
+// manipulator to provision the app credential. The returned credential will have the private key used to generate the CSR
+// added back as an attribute. An error and a nil app cred reference is returned if CSR generation or the API call to the
+// upstream service failed.
+func NewFromTemplate(ctx context.Context, m manipulate.Manipulator, template *gaia.AppCredential) (*gaia.AppCredential, error) {
```

Comment:
It's not really a template. I would call this one `Create()`

Reply:
Done